src/WebForms/UI/WebControls/BaseDataList.cs

```diff
@@ -588,7 +588,8 @@ public static bool IsBindableType(Type type)
         return (type.IsPrimitive ||
                (type == typeof(string)) ||
                (type == typeof(DateTime)) ||
-               (type == typeof(Decimal)));
+               (type == typeof(Decimal)) ||
+               (type == typeof(object)));
```

Comment:
I looked [here ](https://referencesource.microsoft.com/#System.Web/UI/WebControls/BaseDataList.cs,525) , looks like there is no object check.. could you remove ?